docs/guides/deployment/k8s-readiness-liveness.md

```diff
@@ -35,171 +107,693 @@ import fastify from 'fastify'
+2. **Readiness Check** (`/ready` endpoint):
+   - Checks if all Watt services have started successfully
+   - Runs your custom readiness check functions
+   - If successful: pod receives traffic through Kubernetes services
+   - If failed: pod is removed from service endpoints (no restart)
 
-3. **Liveness Check**:
-   - Kubernetes calls the `/status` endpoint every `periodSeconds`
-   - The `watt` server checks that all the services are ready and perform the custom health check for each service
-   - If successful, the container is considered healthy; if it fails `failureThreshold` times, Kubernetes restarts the container
+3. **Liveness Check** (`/status` endpoint):
+   - First verifies readiness (all services started)
+   - Then runs your custom health check functions
+   - If successful: container continues running
+   - If failed: Kubernetes restarts the container
+
+**Important**: The liveness check depends on readiness. If readiness fails, liveness will also fail, potentially causing unnecessary restarts.
 
 ## Project Structure
 
 You can see a full working example in [https://github.com/platformatic/k8s-readiness-liveness](https://github.com/platformatic/k8s-readiness-liveness).
 
 The example project structure demonstrates a Watt application with health checks:
 
-```txt
-├── app
-│   ├── watt.json           # Main Watt configuration
-│   └── services
-│       ├── main            # Entry point service
-│       │   └── platformatic.json
-│       └── service-one     # Example service with custom health check
-│           ├── platformatic.json
-│           └── app.js
-├── k8s
-│   ├── deployment.yaml     # Kubernetes deployment with probes
-│   └── service.yaml        # Kubernetes service configuration
-└── Dockerfile              # Container image build
+```
+├── watt.json               # Main Watt configuration with metrics
+├── Dockerfile              # Container configuration
+├── package.json            # Dependencies and scripts
+├── services/
+│   ├── api/                # Main API service
+│   │   ├── platformatic.json
+│   │   └── index.js        # Service with custom health checks
+│   └── worker/             # Background worker service
+│       ├── platformatic.json
+│       └── index.js
+└── k8s/
+    ├── deployment.yaml     # Kubernetes deployment with probes
+    ├── service.yaml        # Kubernetes service configuration
+    └── ingress.yaml        # Optional ingress configuration
 ```
 
-The `watt.json` configuration exposes the metrics server on port 9090:
+The `watt.json` configuration enables the metrics server:
 
 ```json
 {
   "metrics": {
     "hostname": "{PLT_SERVER_HOSTNAME}",
-    "port": 9090
+    "port": 9090,
+    "readiness": true,
+    "liveness": true
+  },
+  "services": [
+    {
+      "id": "api",
+      "path": "./services/api",
+      "config": "platformatic.json"
+    },
+    {
+      "id": "worker", 
+      "path": "./services/worker",
+      "config": "platformatic.json"
+    }
+  ]
```

Comment:
This is not needed, by default Watt loads all services in the web or services folder via the autoload configuration. Mention this in the guide.